maven-plugin-api/src/main/java/org/apache/maven/plugin/descriptor/MojoDescriptor.java

```diff
@@ -229,20 +227,18 @@ public void addParameter( Parameter parameter )
     }
 
     /**
-     * @return the list parameters as a Map
+     * @return the list parameters as a Map that is built from {@link #parameters} list. Any change to this map is NOT
+     *         reflected on list and other way around!
```

Comment:
It might be worth indicating that the map is keyed by the parameter's `name` ?

Reply:
Fixed